pkg/odo/cli/component/push.go

```diff
@@ -80,6 +80,11 @@ func (po *PushOptions) CompleteDevfilePath() {
 	}
 }
 
+// GetComponentContext gets the component context
+func (po *PushOptions) GetComponentContext() string {
```

Comment:
The componentContext is stored here and is a private variable. This function gets that value and can be used in other CLI packages like storage too.

Reply:
* In `pkg/odo/cli/url`, we are calling this method which is in `pkg/odo/cli/component`. 
* `URLCreateOptions` and `PushOptions` are used interchangeably (as I can see, `GetComponentContext` is in the method set of `PushOptions` struct).
* And as you said, this function gets the value which can be used in other CLI packages as well.

Don't you think that this function belongs in some sort of util package or other top level file for some component instead of being in `pkg/odo/cli/component/push.go`? Besides that, I also feel that it should be removed from the method set of `PushOptions` and put somewhere else. Some kind of common options that are common among various different options viz. `PushOptions`, `URLCreateOptions`, `ComponentOptions`, etc.

Reply:
> Don't you think that this function belongs in some sort of util package or other top level file for some component instead of being in pkg/odo/cli/component/push.go?

Doing that would require a lot of changes related to `componentContext` and the packages where it is being used currently. Thus it should be handled in a separate PR.

Reply:
OK, opened https://github.com/openshift/odo/issues/4340.